test/ads/interstitial_ad_manager_unit.js

```diff
@@ -0,0 +1,325 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+describe('Interstitial Ad manager', () => {
+  const TXml = shaka.util.TXml;
+
+  /** @type {!HTMLElement} */
+  let adContainer;
+  /** @type {!shaka.Player} */
+  let player;
+  /** @type {!shaka.test.FakeVideo} */
+  let mockVideo;
+  /** @type {!jasmine.Spy} */
+  let onEventSpy;
+  /** @type {!shaka.ads.InterstitialAdManager} */
+  let interstitialAdManager;
+
+  beforeEach(() => {
+    adContainer =
+      /** @type {!HTMLElement} */ (document.createElement('div'));
+    player = new shaka.Player();
+    mockVideo = new shaka.test.FakeVideo();
+    onEventSpy = jasmine.createSpy('onEvent');
+    interstitialAdManager = new shaka.ads.InterstitialAdManager(
+        adContainer, player, mockVideo, shaka.test.Util.spyFunc(onEventSpy));
+  });
+
+  afterEach(async () => {
+    interstitialAdManager.release();
+    await player.destroy();
+  });
+
+  it('supports HLS interstitial', async () => {
+    const metadata = {
+      startTime: 0,
+      endTime: null,
```

Comment:
Maybe put DASH and HLS into separate `describe()` suites?

Reply:
Done!